Gems/ROS2/Code/Source/Spawner/ROS2SpawnerInterface.h

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include "ROS2SpawnPointComponent.h"
+#include <AzCore/Component/Entity.h>
+#include <AzCore/Component/EntityId.h>
+#include <AzCore/Math/Transform.h>
+#include <ROS2/Spawner/SpawnerInterface.h>
+
+namespace ROS2
+{
+    class ROS2SpawnerInterface : public ROS2::SpawnerInterface::Registrar
+    {
+    public:
+        const AZ::Transform& GetDefaultSpawnPose() const override;
+
+        // void registerNewSpawnPointComponent(const AZ::EntityId& entityId) override;
+
+        // void removeSpawnPointComponent(const AZ::EntityId& entityId) override;
+        const AZStd::vector<AZStd::pair<AZStd::string, AZStd::shared_ptr<AZ::Transform>>> GetAllSpawnPoints() const;
```

Comment:
The return type shouldn't be a const value since a new instance is returned.
I also I recommend making a type alias for Spawn vector to help document what the `vector<pair<string, shared_ptr<AZ::Transform>>>` is for.
```suggestion
        using SpawnPoint = AZStd::pair<AZStd::string, AZStd::shared_ptr<AZ::Transform>>;
        using SpawnPointVector = AZStd::vector<SpawnPoint>;
        SpawnPointVector GetAllSpawnPoints() const;
```